subprojects/plugins/src/main/java/org/gradle/api/tasks/bundling/War.java

```diff
@@ -166,4 +173,15 @@ public void setWebXml(@Nullable File webXml) {
         this.webXml = webXml;
     }
 
+    /**
+     * Returns the app directory of the task. Defaults to {@code src/main/webapp}.
+     *
+     * @return The app directory.
+     * @since 7.1
+     */
+    @Incubating
+    @Internal
```

Comment:
Why `@Internal` and not `@InputDirectory` in this case?

Reply:
My understanding that `@InputDirectory` assumes the existence of the directory. If the directory is not present the build throws a runtime exception.